src/tablecellproperties/commands/tablecellpropertycommand.js

```diff
@@ -98,4 +95,28 @@ export default class TableCellPropertyCommand extends Command {
 	_getValueToSet( value ) {
 		return value;
 	}
+
+	/**
+	 * Returns a single value for selected table cells. If all cells have the same value set it returns that value or undefined otherwise.
+	 *
+	 * @param {Array.<module:engine/model/element~Element>} tableCell
+	 * @returns {*}
+	 * @private
+	 */
+	_getSingleValue( tableCell ) {
+		const firstCellValue = this._getAttribute( tableCell[ 0 ] );
+
+		const everyCellHasAttribute = tableCell.every( tableCell => this._getAttribute( tableCell ) === firstCellValue );
+
+		return everyCellHasAttribute ? firstCellValue : undefined;
+	}
+}
+
+// Returns all selected table cells.
+function getSelectedTableCells( model ) {
```

Comment:
I'm unsure about this. What if there are two paragraphs in one td? Won't it return this cell twice?

Also, I don't like that there are actually two helper functions introduced in this single PR. Can't we have one, generic enough to cover all those cases? Optionally, two functions `getSelectedTableCell( range )` and `getSelectedTableCells( selection )` next to each other.

Reply:
They have a different purpose so renaming them might be better.

The `getSelectedTableCell` (described below) is used because selection might be collapsed inside table cell or set as a range on a table cell.

The `getSelectedTableCells( selection )` is used to get all selected table cells from a `document.selection` and might be worth to revisit it for the cases found yesteraday for the purpose of `TableSelection#hasMultiRangeSelection`.

ps.: I'm going to revist those later today

Reply:
OK so this one could be removed in favor of `TableSelection.getSelectedTableCells()` - I think that would be OK to use `TableSelection` plugin for this. The `getSelecteTableCells()` can be improved then (ie. in the ticke that touches undo problems).

buuut this would make `TableSelection` obligatory for `TableProperties` which I'm not sure how to handle.

Probably a generally accessible `util` method which all commands/plugins could use to get selected table cells (I'm thinking about other commands that should be now aware of the new option of selecting table cells (collapsed selection in TC or multi-range selection on TCs).

This way we will be having one way to get selected TC as an util (independent of loaded plugins) which will work both cases of selection. This could also invalidate need for `getSelectedTableCell()` which returns only one - you can always get one from all selected.